### ISSUE TITLE
PJAX form submit and yii confirm don't mix.

### ISSUE THREAD

**Issue by pcfreak30** [yiisoft/yii2#6363]
Right now if you use ActionColumn and set data-pjax to 1, then it doesn't do an ajax submit like in v1.1.

The yii.js file needs to be modified so that ```$form = $('<form method="' + method + '" action="' + action + '"></form>');``` has ```data-pjax```.

The form needs to be inserted as well after the called element, not the body, so change ```$form.hide().appendTo('body');``` to ```$e.after($form.hide());```

This is so the jquery selector defined in Pjax.php on line 175 get matched.

There also should be a better way to handle this as you must define pjax history to be off, or the replaceState js function does a window.location.reload().

In summary, the ActionColumn should support ajax actions via some means. This was a mess to figure out...

**Comment by germansokolov13** [comment]
Same problem happens to me. `data-confirm` attribute on a link doesn't let pjax-call to fire in a gridview. After you click OK in a browser-provided confirm dialog, full page reload occurs. If you remove `data-confirm` it starts to work via pjax.
